### PR TITLE
test: emit info diagnostic from `st.css`

### PR DESCRIPTION
**Pull request by tzachbon**

### CODE REVIEW

File: packages/cli/test/cli.spec.ts [hunk 1]
```diff
@@ -268,12 +268,10 @@ describe('Stylable Cli', function () {
...
         });
 
-        it.skip('(diagnosticsMode) should not exit with error when using strict mode with only info diagnostics', () => {
-            // Todo: test info diagnostic when we have one.
-            // https://github.com/wix/stylable/pull/2018
+        it('(diagnosticsMode) should not exit with error when using strict mode with only info diagnostics', () => {
             populateDirectorySync(tempDir.path, {
                 'package.json': `{"name": "test", "version": "0.0.0"}`,
-                'style.st.css': `.root {} `,
+                'style.st.css': `:vars { colors: stArray(red, blue); } `,
```
Comment (tomrav) [inline packages/cli/test/cli.spec.ts:274]:
```suggestion
                'style.st.css': `:vars { colors: stArray(red, blue); }`,
```

File: packages/cli/test/cli.spec.ts [hunk 2]
```diff
@@ -285,6 +283,7 @@ describe('Stylable Cli', function () {
             expect(status).to.equal(0);
             expect(stdout, 'stdout').to.match(/\[Stylable Diagnostics\]/);
             expect(stdout, 'stdout').to.match(/style\.st\.css/);
+            expect(stdout, 'stdout').to.match(/\[info\]: "stArray" is deprecated, use "st-array"/);
```
Comment (tomrav) [inline packages/cli/test/cli.spec.ts:286]: use diagnostic function?

File: packages/webpack-plugin/test/e2e/emit-info-diagnostic.spec.ts [hunk 3]
```diff
@@ -26,7 +26,7 @@ describe(`(${project})`, () => {
         const warnings = projectRunner.getBuildWarningMessages();
 
         expect(warnings).to.have.length(1);
-        expect(warnings[0]).to.match(/test info warning!/);
+        expect(warnings[0]).to.match(/"stArray" is deprecated, use "st-array"/);
```
Comment (tomrav) [inline packages/webpack-plugin/test/e2e/emit-info-diagnostic.spec.ts:29]: use diagnostic function?

File: packages/webpack-plugin/test/e2e/projects/emit-info-diagnostic/webpack.config.js [hunk 4]
```diff
@@ -9,17 +9,6 @@ module.exports = {
     plugins: [
         new StylableWebpackPlugin({
             diagnosticsMode: 'strict',
```
Comment (tomrav) [inline packages/webpack-plugin/test/e2e/projects/emit-info-diagnostic/webpack.config.js:11]: this is here to show that it doesn't break in strict mode with info diagnostics?

**Review by tomrav — COMMENTED** [review]
Left a couple of questions

File: packages/webpack-plugin/test/e2e/projects/emit-info-diagnostic/webpack.config.js (hunk 4 above)
Comment (tzachbon) [reply packages/webpack-plugin/test/e2e/projects/emit-info-diagnostic/webpack.config.js:11]: yep

**Review by barak007 — APPROVED** [review]
I know we sad that we want to check it on a real info diagnostics but there is a downside for that.
When we will get no more infos we will not have something to test the behavior.

It's not a blocker for these tests. but the approach of emitting custom info seems more resilient to me.

**Comment by tzachbon** [comment]
> I know we sad that we want to check it on a real info diagnostics but there is a downside for that.
> When we will get no more infos we will not have something to test the behavior.
> 
> It's not a blocker for these tests. but the approach of emitting custom info seems more resilient to me.

@barak007 alright then, we can wait with this until we have permanent info diagnostic or a way to hook to the post-process of the CLI #1813

**Comment by tzachbon** [comment]
> I know we sad that we want to check it on a real info diagnostics but there is a downside for that.
> When we will get no more infos we will not have something to test the behavior.
> 
> It's not a blocker for these tests. but the approach of emitting custom info seems more resilient to me.

@barak007 I decided to keep the post-process hook until we will have a permanent `info` diagnostic.

**Review by barak007 — APPROVED** [review]

**Pull request merged**